### PR TITLE
Add Google 2020 Root Power writeup

### PR DESCRIPTION
**Pull request by nankeen**

### CODE REVIEW

File: _posts/2020-08-25-googlectf-2020-root-power.markdown [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+```bash
+mkdir -p mnt && sudo mount -o loop,offset=1048576 disk.img mnt
+```
+
+My first instinct was to check `/etc/shadow` and the boot files, the former was a rabbit hole.
+I also checked `initramfs` to see if there are any files I might've missed, it contains `ssdt.aml`, we'll come back to this later.
+
+## Authentication mechanism
+A feature that authenticated users in Linux was PAM.
```
Comment (Josh-00FF00) [inline _posts/2020-08-25-googlectf-2020-root-power.markdown]:
```suggestion
Users are authenticated by the PAM service in Linux
```

File: _posts/2020-08-25-googlectf-2020-root-power.markdown [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  
+  fd = fopen("/dev/chck","r");
+  fgets(buf,2,fd);
+  fclose(fd);
+  ret = atoi(buf);
+  return ret;
+}
+```
+
+What exactly provides this device?
```
Comment (Josh-00FF00) [inline _posts/2020-08-25-googlectf-2020-root-power.markdown:110]:
```suggestion
What does this device provide exactly?
```

File: _posts/2020-08-25-googlectf-2020-root-power.markdown [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+
+## Chck kernel module
+
+Devices in `/dev/` are usually added through kernel modules, so I looked for `.ko` files in the image.
```
Comment (Josh-00FF00) [inline _posts/2020-08-25-googlectf-2020-root-power.markdown]: I'm not sure "_usually_" is true, but I'm being nitpicky. Most of the stuff in `/dev/` are tty and other core kernel stuff as opposed to module additions. But I get yer point

File: _posts/2020-08-25-googlectf-2020-root-power.markdown [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+This function relays reads on `/dev/chck` to ACPI's _CHCK_ device.
+My guess was that this was handled early in the boot process --- `initramfs`.
+
+## AML file
+
+`initramfs` contains `ssdt.aml` which can be decompiled with `iasl`, producing ssdt.dsl.
+The following helped me make sense of what is happening:
+
+1. https://uefi.org/specifications
```
Comment (Josh-00FF00) [inline _posts/2020-08-25-googlectf-2020-root-power.markdown]: Make these links?

File: _posts/2020-08-25-googlectf-2020-root-power.markdown [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+* `KBDA` contained a bunch of pre-defined bytes.
+
+What kind of device is this? Googling ___acpi 0x60 0x64___ yields [this page](https://wiki.osdev.org/%228042%22_PS/2_Controller).
+So I made the assumption that `KBDA` contained __PS/2 scan codes__.
+
+That assumption was confirmed with [this table](http://www.vetra.com/scancodes.html).
+Each byte represents a make/break code, indicating a keydown/keyup event.
+
+The decode script is trivial and is left as an exercise for the reader.
+_Lol kidding_, I did it by hand.
```
Comment (Josh-00FF00) [inline _posts/2020-08-25-googlectf-2020-root-power.markdown:173]: Nice 😎

File: _posts/2020-08-25-googlectf-2020-root-power.markdown (hunk 3 above)
Comment (nankeen) [reply _posts/2020-08-25-googlectf-2020-root-power.markdown]: Ah yeah you're right, I think _"__Additional__ devices in `/dev/` are usually added through..."_ was what I meant.

File: _posts/2020-08-25-googlectf-2020-root-power.markdown (hunk 4 above)
Comment (nankeen) [reply _posts/2020-08-25-googlectf-2020-root-power.markdown]: Didn't notice jekyll didn't do that automatically

**Pull request merged**